apps/system/js/accessibility.js

```diff
@@ -576,35 +578,41 @@
      */
     cancelHints: function ar_cancelHints() {
       clearTimeout(this.hintsTimer);
-      if(this.isSpeakingHints){
-        this.cancelSpeech();
-        this.isSpeakingHints = false;
-      }
+      this.cancelSpeech();
     },
 
     /**
-     * Based on whether the screen reader is currently enabled, announce the
-     * instructions of how to enable/disable it.
+     * Based on whether the screen reader is currently enabled, announce a
+     * message such as instructions of how to enable/disable it.
      * @memberof Accessibility.prototype
      */
-    announceScreenReader: function ar_announceScreenReader() {
+    announceScreenReader: function ar_announceScreenReader(
```

Comment:
TBH: I would rather have the logic outside of this function. So when announceScreenReader is called, check enabled and pass an appropriate key.